packages/compiler/src/core/checker.ts

```diff
@@ -328,15 +336,10 @@ const TypeInstantiationMap = class
   extends MultiKeyMap<readonly Type[], Type>
   implements TypeInstantiationMap {};
 
-let currentSymbolId = 0;
-
-export function createChecker(program: Program): Checker {
+export function createChecker(program: Program, resolver: NameResolver): Checker {
   const stdTypes: Partial<StdTypes> = {};
-  const symbolLinks = new Map<number, SymbolLinks>();
-  const mergedSymbols = new Map<Sym, Sym>();
   const docFromCommentForSym = new Map<Sym, string>();
   const augmentDecoratorsForSym = new Map<Sym, AugmentDecoratorStatementNode[]>();
```

Comment:
I think we could move this whole resolution into the name resolver + consider renaming it to linker